### PR TITLE
feat(ScoobieLink): Render download links as anchors

### PR DESCRIPTION
**Pull request by raphsutti**
This adds a third case to custom links behaviour. 

Previously we account for internal and external links based on `href` attribute. Here we use `download` attribute to determine if the link is for download and not to be navigated to (no client side rendering happens)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -391,6 +391,8 @@ Render all underlying links as follows:
 - Internal links use client-side navigation with smooth scrolling via [react-router-hash-link],
   and pass through the `v` URL parameter for UI version switching
 - External links open in a new tab
+- Attachment links open attachments in a new tab.
```
Comment (72636c) [inline README.md:394]: Do we want this to open in a new tab? If not, we can unset the `target` in this scenario.

File: README.md [hunk 2]
```diff
@@ -391,6 +391,8 @@ Render all underlying links as follows:
...
+  Link components must set `datatype` attribute to `attachments`
```
Comment (72636c) [inline README.md:395]: 1. What's `datatype`? A cursory search seems to tell me that this is a non-standard attribute that has no usage internally at SEEK.
2. Would this be better as a singular `attachment` or maybe even `download`?

File: README.md (hunk 1 above)
Comment (raphsutti) [reply README.md:394]: I chose to open in new tab because when opening pdf links, you end up completely leaving the site

File: README.md (hunk 2 above)
Comment (raphsutti) [reply README.md:395]: 1. Couldn't find nice attributes to use that are typed in `makeLinkComponent` 
2. Yep download sounds good. But just noting that for pdf case, the pdf opens in browser so doesn't feel download-y

Comment (72636c) [reply README.md:395]: How about this: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download

File: src/components/ScoobieLink.tsx [hunk 3]
```diff
@@ -6,14 +6,33 @@ import { isExternalHref } from '../private/url';
...
+        >
+          {children}
+        </a>
+      );
+    }
+
+    if (download) {
+      return (
+        <a rel="noreferrer" {...restProps} href={href} ref={ref}>
+          {children}
```
Comment (raphsutti) [inline src/components/ScoobieLink.tsx:27]: Might not be too bad but not sure how I can clean up this up. Repeating itself a little here. 

The externalHref anchor tag has target attribute whereas this download anchor tag doesnt

File: README.md (hunk 1 above)
Comment (72636c) [reply README.md:394]: Is this because you aren't passing through the `download` prop to `a`?

**Review by 72636c — COMMENTED** [review]
Nice one! Can we update the README and PR title?

File: README.md (hunk 1 above)
Comment (raphsutti) [reply README.md:394]: Ah, failed

Comment (raphsutti) [reply README.md]: I've amended this to open in the same tab

File: README.md [hunk 4]
```diff
@@ -391,6 +391,9 @@ Render all underlying links as follows:
 - Internal links use client-side navigation with smooth scrolling via [react-router-hash-link],
   and pass through the `v` URL parameter for UI version switching
 - External links open in a new tab
+- Download links prompt user to download attachments.
+  Link components must set `download` attribute and a suggested file name value may be provided.
+  If no value is provided, a default file name will be given.
```
Comment (72636c) [inline README.md:396]:
```suggestion
- Links with a [`download` attribute] prompt the user with a file download

[`download` attribute]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download
```

**Review by 72636c — APPROVED** [review]

**Pull request merged**